silkworm/common/tee.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+   Copyright 2020 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#ifndef SILKWORM_COMMON_TEE_H_
+#define SILKWORM_COMMON_TEE_H_
+
+#include <streambuf>
+#include <iostream>
+#include <fstream>
+#include <ctime>
+#include <cstdio>
+
+namespace silkworm {
+
+// Tee code adapted from http://wordaligned.org/articles/cpp-streambufs
+class teebuf: public std::streambuf {
+public:
+    // Construct a streambuf which tees output to the supplied streambufs.
+    teebuf(std::streambuf* sb1, std::streambuf* sb2)
+      : sb1(sb1), sb2(sb2) {}
+
+private:
+    // This tee buffer has no buffer. So every character "overflows"
+    // and can be put directly into the teed buffers.
+    virtual int overflow(int c) {
```

Comment:
Let's say override instead of virtual ([Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Inheritance))